core/trino-main/src/main/java/io/trino/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -783,7 +783,7 @@ private FormattedDomain parseDomain(Domain domain)
                                     .collect(toImmutableSet())),
                     discreteValues -> formattedRanges.addAll(
                             discreteValues.getValues().stream()
-                                    .map(value -> valuePrinter.castToVarcharOrFail(type, value))
```

Comment:
Now this `castToVarcharOrFail` is potentially unused right ? Can we remove that method or make it private ?